Add Python 3.12 support

pbratkowski opened the pull request:
* Add Python 3.12 to test matrix in ci.yml and tox.ini

* Add Python 3.12 classifier to setup.py

* Update lxml dependency from 4.9.2 to 4.9.3

Notes:

- Python 3.12 fails to install `lxml` 4.9.2, so I updated the dependency.
- Django 4.2 does not officially support Python 3.12 until version 4.2.8, so it will need to be added to the test matrix at a later date.

pbratkowski committed:
Add Python 3.12 support

* Add Python 3.12 to test matrix in ci.yml and tox.ini

* Add Python 3.12 classifier to setup.py

* Update lxml dependency from 4.9.2 to 4.9.3

carltongibson commented:
Looks the same as #1208

pbratkowski commented:
> Looks the same as #1208

You're right. We were adding adding Django 5.0 support over on Discord and figured we should add py 3.12 too. Sorry, I should have checked if there were any open PRs before opening this one :)

Also, it fails to install due to `distutils` being deprecated, which was not an issue when running `tox` locally. Closing this for now.

carltongibson commented:
Yes the distutil usage will need to be removed. That's the remaining issue here (if you're feeling up to it)

Thanks for the input!

pbratkowski commented:
> Yes the distutil usage will need to be removed. That's the remaining issue here (if you're feeling up to it)

Sure thing. I'll need to open a new PR though as GH won't let me reopen this (possibly because I force pushed). How would this look: https://github.com/django-compressor/django-compressor/compare/develop...pbratkowski:django-compressor:py312?expand=1

I `cherry-pick`ed 3 commits from #1208, updated the `ci.yml` and `tox.ini` files and added the classifiers, so foarsitter gets credit and I don't need to rewrite their code, and all changes are applied on top of Tim's latest contribution. It would also make #1210 redundant I think.
